[TOKO] Wise Wallet - week3 1st PR

aaaz425 opened the pull request:
# 구현 미리보기

<img src="https://user-images.githubusercontent.com/101464713/226887641-516ca769-81d1-4cc6-83f6-bdc9720bef9b.mov">

# 커밋 요약 및 한 줄 소감

- 2주차 까지 구현한 내용을 정리한다
- document 전체에 이벤트를 걸기 보다는 최대한 하위 요소에 이벤트를 걸 수 있도록 한다 -> 이벤트 상속이라는 개념을 공부한다
- 처음 코드를 짤 때부터 변수명에 의도를 드러낼 수 있도록 노력한다
- 함수는 더 이상 쪼개지지 않을 때 까지 분리한다 -> but 지금은 내가 감당 가능 한 정도 까지만 하자
- if 문 의 조건에 블리언 데이터를 활용해보자 -> true 나 false 값을 의도가 드러나는 변수명에 저장!
- local storage에 어떤 데이터를 어떻게 저장하고 어떻게 활용할지 구상한다

Commit:
Docs : 2주차 구현 내용 업데이트

README 파일에 2주차 구현 내용을 업데이트 하고 체크리스트를 보강한다

Commit:
Feat : input 요소 autocomplete 기능 비활성화 및 중복된 css 속성 중복제거

메인 페이지의 input 요소 autocomplete 기능을 비활성화 시키고 중복된 css 속성 중복 제거

Commit:
Chore : 폴더 삭제 및 변경

폴더를 삭제하고 경로를 변경한다

Commit:
Style : 모든 파일에 prettier 적용

프로젝트 내 모든 파일에 prettier를 적용한다

Commit:
Feat : 메인 페이지 인풋바 체크버튼 활성화 기능 구현 중

메인 페이지의 인풋바안에 input 요소의 value가 모두 1 이상 일 때 체크 버튼이 활성화 되도록 하는 기능을 구현한다 / 미완성

namse commented on src/js/input_bar/observer.js:
```diff
@@ -0,0 +1,41 @@
+$inputs.forEach(ele => ele.addEventListener("input", () => {
+  let len = 0;
+  $inputs.forEach(v => {
+    v.value.length > 0 ? len += 1 : len;
```
제발 if 문 그냥 써줘요잉

namse commented on src/js/input_bar/observer.js:
```diff
@@ -0,0 +1,41 @@
+  if (len === 5) {
+    $checkBtn.setAttribute("src", "./src/svg/activate_check.svg");
+    $checkBtn.setAttribute("condition", "activate");
+    len = 0;
```
len = 0 왜!?

namse commented on src/js/input_bar/observer.js:
```diff
@@ -0,0 +1,41 @@
+  $inputs.forEach(v => {
+    v.value.length > 0 ? len += 1 : len;
+  });
+  if (len === 5) {
```
if (len === 5) 만 봤을때 코드 읽는 사람이 무슨 의도로 이걸 썼는지 파악하기가 어렵다. 그쵸잉?
그렇다면 어떡해야할까요? 주석을 쓴다? 흠... 그건 쪼옴~? 어떻게 하면 좋을까! 생각해보기.

namse replied:
`const isActive = len === 5;` 라고 힌트 아닌 답을 드림

Commit:
Feat : 메인 페이지 인풋바 체크 버튼 활성화 기능 추가

메인 페이지 인풋바의 체크 버튼에 이벤트 리스너를 추가하여 모든 input창의 길이가 1 이상일 때 체크 버튼이 활성화 될 수 있게 한다

Commit:
Feat : 의도를 나타낼 수 있게 변수명을 정하고 함수를 분리

변수명만 보고도 의도를 어느정도 알 수 있도록 변수명을 수정하고, 함수를 최대한 분리한다, if문 조건 안에 블리언 데이터를 이용한다

Commit:
Feat : input bar의 수입/지출 상태를 변경하는 버튼을 눌렀을 때 date를 제외한 input 데이터를 초기화

메인 페이지 input bar의 수입/지출 상태를 변경하는 체인지 버튼을 눌렀을 때, date를 제외한 input의 데이터 값을 빈 문자열로 초기화한다

Commit:
Feat : 체크 버튼이 활성화 되어있을 때 호버 기능 추가

인풋바의 체크 버튼이 활성화 되어 있을 때 호버 기능을 추가하여 이미지의 크기를 1.2배 크게 한다

hjsong333 commented on src/js/input_bar/input_list.js:
```diff
@@ -0,0 +1,119 @@
+      "true";
+    const newDropdown = document.createElement("div");
+    if (isMinus) {
+      this.expenseCategories.forEach((category) => {
+        this.addItem(category, newDropdown);
+      });
+    } else {
+      this.incomeCategories.forEach((category) => {
+        this.addItem(category, newDropdown);
+      });
+    }
+    this.categoryDropdownMenu.appendChild(newDropdown);
+  },
+
+  showPaymentDropdownMenu() {
+    this.clearDropdownMenu(this.paymentDropdownMenu);
+    this.addPaymentDropdownMenu();
+    this.showDropdownMenu(this.paymentDropdownMenu);
+  },
+
+  showCategoryDropdownMenu() {
+    this.clearDropdownMenu(this.categoryDropdownMenu);
+    this.addCategoryDropdownMenu();
+    this.showDropdownMenu(this.categoryDropdownMenu);
+  },
+
+  handlePaymentDropdownMenuClick() {
+    this.paymentDropdownMenu.addEventListener("click", (e) => {
+      this.paymentInputField.value = "";
+      this.paymentInputField.value = e.target.textContent;
+    });
+  },
+
+  handleCategoryDropdownMenuClick() {
+    this.categoryDropdownMenu.addEventListener("click", (e) => {
+      this.categoryInputField.value = "";
+      this.categoryInputField.value = e.target.textContent;
+    });
+  },
+};
```
이렇게 객체로 묶는 이유가 있을까요?

aaaz425 replied:
단순히 전역변수를 없애기 위함 이었습니다 ㅎㅎ
현재는 변수만을 만들어 놓은 파일을 따로 만들어 모듈화 시키는 중입니다🙌

hjsong333 commented on src/js/input_bar/input_bar_btn.js:
```diff
@@ -0,0 +1,47 @@
+    if (isfilled) {
+      activateCount += 1;
+    } else {
+      activateCount;
```
변수에 아무 작업도 해주지 않는 것 같습니다.

aaaz425 replied:
삼항연산자를 if 문으로 바꾸면서 손가락의 흐름대로 하다보니 
불필요한 else 구문이 생긴 것 같습니다 ㅎㅎ

hjsong333 commented:
https://github.com/codesquad-members-2023/fe-max--wise-wallet/pull/44#discussion_r1145758653

hjsong333 commented on index.html:
```diff
@@ -27,12 +27,13 @@
-    <script defer src="./src/js/events/input_bar/input_date.js"></script>
-    <script defer src="./src/js/events/header/header_monthYear.js"></script>
-    <script defer src="./src/js/events/header/header_tab.js"></script>
+    <script defer src="./src/js/input_bar/datePicker.js"></script>
+    <script defer src="./src/js/input_bar/input_list.js"></script>
+    <script defer src="./src/js/input_bar/input_price.js"></script>
+    <script defer src="./src/js/input_bar/input_date.js"></script>
+    <script defer src="./src/js/header/header_monthYear.js"></script>
+    <script defer src="./src/js/header/header_tab.js"></script>
+    <script defer src="./src/js/input_bar/input_bar_btn.js"></script>
```
defer를 쓰신 이유는 무엇인가요?

aaaz425 replied:
각 js 파일에서 ES6 모듈을 사용 하지 않고
html 헤드 태그에서 일일이 불러오느라 defer를 사용 했습니다
지금 모듈에 대해 학습중이어서 학습이 끝난 후 지저분한 script를 정리할 수 있을 것 같습니다😃
